gokrazy.go

```diff
@@ -223,7 +230,17 @@ func Supervise(commands []*exec.Cmd) error {
 		return err
 	}
 
-	if err := updateListenerPairs("80", "443", useTLS, tlsConfig); err != nil {
+	httpPort, err := readFile("http-port.txt")
+	if _, numeric := strconv.Atoi(httpPort); err != nil || numeric != nil {
+		httpPort = "80"
```

Comment:
Here and below: please log an error message when the file is present but malformed, so that users know what the problem is.

Reply:
Split into function as the validation with logging is more than 2 lines